third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/DotOpToLLVM/MMAv2.cpp

```diff
@@ -427,7 +431,7 @@ LogicalResult convertDot(const LLVMTypeConverter *typeConverter,
   auto batchOffset =
       elemsPerThread[rank - 2] * elemsPerThread[rank - 1] / numCPackedElem;
   auto callMma = [&](unsigned b, unsigned m, unsigned n, unsigned k) {
-    unsigned colsPerThread = repN * 2;
+    unsigned colsPerThread = repN > 1 ? repN * 2 : repN;
```

Comment:
any chance we can fix this special casing, now that we are at it?

Reply:
Actually it's not a special case. It should always be `repN * 2`.